dataflow/src/domain/mod.rs

```diff
@@ -1389,11 +1398,20 @@ impl Domain {
                                 let ref n = *nd.borrow();
                                 let local_index: LocalNodeIndex = *n.local_addr();
 
-                                self.state
-                                    .get(&local_index)
-                                    .filter(|state| state.is_partial())
-                                    .map(|state| state.deep_size_of())
-                                    .unwrap_or(0)
+                                if n.is_reader() {
+                                    // We are a reader, which has its own kind of state
+                                    let mut size = 0;
+                                    n.with_reader(|r| size = r.state_size().unwrap_or(0))
+                                        .unwrap();
+                                    size
+                                } else {
+                                    // Not a reader, state is with domain
+                                    self.state
+                                        .get(&local_index)
+                                        .filter(|state| state.is_partial())
```

Comment:
So reported memory use is only the size of partial states then?

Reply:
I don't see us making the same distinction from `Reader`s, even though presumably the same should apply to them?

Reply:
`UpdateStateSize` is only used for eviction, and there's no point in evicting from anything other than a partial state.

`GetStatistics` includes all states.

Reply:
Good point re readers, will fix.

Reply:
27cb886dfd3fba9245ca3e3bed231816480db369